singularity-eos/eos/eos_davis.hpp

```diff
@@ -354,19 +380,21 @@ PORTABLE_INLINE_FUNCTION Real DavisReactants::Es(const Real rho) const {
   } else {
     e_s = -y - (1.0 - std::exp(b4y)) / (4.0 * _B);
   }
-  return _e0 + _P0 * (1.0 / _rho0 - 1.0 / rho) + phat / _rho0 * e_s;
+  return _e0 + _P0 * (1.0 / _rho0 - robust::ratio(1.0, std::max(rho, 0.))) +
+         phat / _rho0 * e_s;
 }
 PORTABLE_INLINE_FUNCTION Real DavisReactants::Ts(const Real rho) const {
   if (rho >= _rho0) {
-    const Real y = 1 - _rho0 / rho;
-    return _T0 * std::exp(-_Z * y) * std::pow(_rho0 / rho, -_G0 - _Z);
+    const Real y = 1 - robust::ratio(_rho0, std::max(rho, 0.));
+    return _T0 * std::exp(-_Z * y) *
+           std::pow(robust::ratio(_rho0, std::max(rho, 0.)), -_G0 - _Z);
   } else {
-    return _T0 * std::pow(_rho0 / rho, -_G0);
+    return _T0 * std::pow(robust::ratio(_rho0, rho), -_G0);
```

Comment:
In this branch where rho could go to zero, it might make sense to make sure rho is nonnegative.  In the end, you might just want to store `robust::ratio(_rho0.,std::max(rho,0.0))` at the beginning of this routine since it is re-used multiple times.

Reply:
For this comment and the previous, I had eliminated negative density branches in favor of just forcing a positive density. I'll do that here too